### PR TITLE
refactor: include det-deploy into det cli. [DET-5153]

### PR DESCRIPTION
**Pull request by ioga**
## Checklist

- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](../docs/release-notes/README.md) for details.

[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234

### CODE REVIEW

File: docs/how-to/installation/deploy.txt [hunk 1]
```diff
@@ -1,11 +1,11 @@
 .. _install-using-deploy:
 
 #####################################
- Install Determined Using det-deploy
+ Install Determined Using det deploy
```
Comment (shiyuann) [inline docs/how-to/installation/deploy.txt:4]:
```suggestion
 Install Determined Using ``det deploy``
```

File: harness/determined/deploy/cli.py [hunk 2]
```diff
@@ -1,40 +1,50 @@
...
+from .local.cli import args_description as local_args_description
 
-def main() -> None:
-    environment_map = {
-        "aws": determined.deploy.aws.cli.deploy_aws,
-        "gcp": determined.deploy.gcp.cli.deploy_gcp,
-        "local": determined.deploy.local.cli.deploy_local,
-    }
+args_subs: List[Union[Arg, Cmd]] = [
+    # TODO(ilia): Remove --version flag when det-deploy is deprecated.
```
Comment (shiyuann) [inline harness/determined/deploy/cli.py:13]: can you create a ticket for that and put the ticket number here rather than your name?

**Review by shiyuann — COMMENTED** [review]
Looks good! Have you tested those commands?

File: docs/how-to/installation/deploy.txt (hunk 1 above)
Comment (ioga) [reply docs/how-to/installation/deploy.txt]: 👍

File: harness/determined/deploy/cli.py (hunk 2 above)
Comment (ioga) [reply harness/determined/deploy/cli.py]: done

**Review by shiyuann — APPROVED** [review]
Looks good!

**Pull request merged**